ocs_ci/ocs/pgsql.py

```diff
@@ -663,3 +664,193 @@ def pgsql_full(self):
+        postgres_pod,
+        table_name,
+        total_rows,
+        num_of_ops,
+        run_time,
+        drop_table=True,
+    ):
+        """
+        Run queries on the Postgres database,
+        The steps include:
+        1. Create table <table_name> and insert <total_rows> number of records.
+        2. Repeat below steps in order for specified <run_time> duration:
+        a. Insert <num_of_ops> number of rows into the table.
+        b. Update <num_of_ops> number of rows into the table.
+        c. Update random number of rows based on matching year in the date column of records.
+        d. Delete <num_of_ops> number of rows from the table.
+
+        Args:
+            postgres_pod (POD) : POD object on which postgres is running.
+            table_name (str) : Name of the table that we want to create.
+            total_rows (int) : Total number of records that should be there inside the table.
+            num_of_ops (int) : Total number of records we want to be inserted/deleted/updated in each operation
+                                respectively.
+            run_time (int) : Total Run Time in minutes.
+            drop_table (bool) : True if we want to delete the table, False otherwise.
+
+        Returns:
+             str  : A string containing the last valid state of the table.
+
+        """
+        row_index = 0
+        last_valid_state = ""
+        self.run_pgsql_command(
+            postgres_pod,
+            f"CREATE TABLE {table_name} (row_id INT PRIMARY KEY, username VARCHAR (50) NOT NULL, date DATE NOT NULL);",
+        )
+        self.run_insert_operation(total_rows, postgres_pod, 0, table_name)
+        row_index += total_rows
+        self.run_pgsql_command(postgres_pod, f"SELECT * FROM {table_name};", True)
+
+        end_time = datetime.now() + timedelta(minutes=run_time)
+        while datetime.now() < end_time:
+            for sql_operation in range(0, 4):
+                if sql_operation == 0:
+                    self.run_insert_operation(
+                        num_of_ops, postgres_pod, row_index, table_name
+                    )
+                    row_index += num_of_ops
+                elif sql_operation == 1:
+                    self.run_update_operation(
+                        num_of_ops, postgres_pod, row_index, table_name
+                    )
+                elif sql_operation == 2:
+                    self.run_pgsql_command(
+                        postgres_pod,
+                        (
+                            f"UPDATE {table_name} SET date='{self.generate_random_date()}' "
+                            f"WHERE EXTRACT(YEAR FROM date) = {str(random.randint(1900, datetime.now().year))};"
+                        ),
+                    )
+                else:
+                    self.run_delete_operation(
+                        num_of_ops, postgres_pod, row_index, table_name
+                    )
+                last_valid_state = self.run_pgsql_command(
```

Comment:
Since the last operation before this query is delete the row you add, this will return  an empty value, of error.
it should be in the loop before the delete.
also, it will be better to log the output which return.

Reply:
here we are not deleting all the row. Only the num_of_ops number of rows will be deleted from the table. So last_valid_state will never be empty at this line. last_valid_state captures the snapshot of the last valid state of the table, this can be used for checking the data integrity after certain operations or failures.